tools/base/fpgad/plugins/fpgad-vc/fpgad-vc.c

```diff
@@ -500,40 +505,145 @@ STATIC fpga_result vc_unload_driver(vc_device *vc)
 
 	fpgaDestroyProperties(&prop);
 
-	//           11111111112222222222333
-	// 012345678901234567890123456789012
-	// /sys/bus/pci/devices/ssss:bb:dd.f
+	snprintf_s_iiii(path, PATH_MAX,
+			"/sys/bus/pci/devices/%04x:%02x:%02x.%d",
+			(int)seg, (int)bus, (int)dev, (int)fn);
+
+	memset_s(rlpath, sizeof(rlpath), 0);
 
-	snprintf_s_i(path, SYSFS_PATH_MAX,
-		     "/sys/bus/pci/devices/%04x:",
-		     (int)seg);
+	if (readlink(path, rlpath, sizeof(rlpath)) < 0) {
+		LOG("readlink \"%s\" failed.\n", path);
+		return FPGA_EXCEPTION;
+	}
 
-	snprintf_s_i(&path[26], SYSFS_PATH_MAX - 26,
-		     "%02x:", (int)bus);
+	// (rlpath)
+	//
+	// ../../../devices/pci0000:ae/0000:ae:00.0/0000:af:00.0/
+	// 0000:b0:09.0/0000:b2:00.0
 
-	snprintf_s_i(&path[29], SYSFS_PATH_MAX - 29,
-		     "%02x.", (int)dev);
+	err = strstr_s(rlpath, sizeof(rlpath),
+		       "devices", 7, &p);
+	if (err != EOK) {
+		LOG("error: no \"devices\" in path \"%s\"\n", rlpath);
+		return FPGA_EXCEPTION;
+	}
 
-	snprintf_s_i(&path[32], SYSFS_PATH_MAX - 32,
-		     "%d/remove", (int)fn);
+	// /sys/devices/pci0000:ae/0000:ae:00.0/0000:af:00.0/
+	// 0000:b0:09.0/0000:b2:00.0/reset
+	snprintf_s_s(path, PATH_MAX,
+		      "/sys/%s/reset", p);
 
-	LOG("writing 1 to %s to remove driver.\n", path);
-	if (file_write_string(path, "1\n", 2)) {
+	LOG("writing 1 to \"%s\" to reset the device.\n", path);
+	res = file_write_string(path, "1\n", 2);
```

Comment:
Please pick up the latest from master.  This code is re-introducing a bug.

Reply:
I just did a cherry pick from master. Is it not correct?

Reply:
I see. You made a new check-in a few minutes ago.